src/ray/raylet/node_manager.cc

```diff
@@ -1933,79 +1933,127 @@ void NodeManager::FinishAssignedActorTask(Worker &worker, const Task &task) {
-          /*failure_callback=*/
-          [](gcs::AsyncGcsClient *client, const ActorID &id, const ActorTableData &data) {
-            // Only one node at a time should succeed at creating the actor.
-            RAY_LOG(FATAL) << "Failed to update state to ALIVE for actor " << id;
-          });
-    }
+    auto new_actor_data = CreateActorTableDataFromCreationTask(task);
+    // Lookup the parent actor id
+    auto parent_task_id = task.GetTaskSpecification().ParentTaskId();
+    RAY_CHECK_OK(gcs_client_->raylet_task_table().Lookup(
+        JobID::Nil(), parent_task_id,
+        //success_callback
+        [this, task, actor_id, actor_handle_id, new_actor_data, resumed_from_checkpoint]
+           (ray::gcs::AsyncGcsClient *client, const TaskID &parent_task_id,
+               const TaskTableData &parent_task_data) mutable {
+          // The task was in the GCS task table. Use the stored task spec to
+          // get the parent actor id.
+          auto message = flatbuffers::GetRoot<protocol::Task>(parent_task_data.task().data());
+          Task parent_task(*message);
+          new_actor_data.set_parent_actor_id(parent_task.GetTaskSpecification().ActorId().Binary());
+          FinishAssignedActorTaskHelper(actor_id, new_actor_data, resumed_from_checkpoint);
+          if(!resumed_from_checkpoint){
+            // The actor was not resumed from a checkpoint. We extend the actor's
+            // frontier as usual since there is no frontier to restore.
+            ExtendActorFrontier(task, actor_id, actor_handle_id);
+          }
+        },
+        //failure_callback
+        [this, task, actor_id, actor_handle_id, new_actor_data, resumed_from_checkpoint]
+            (ray::gcs::AsyncGcsClient *client, const TaskID &parent_task_id) mutable {
+          // The parent task was not in the GCS task table. It must therefore be in the
+          // lineage cache.
+          RAY_CHECK(lineage_cache_.ContainsTask(parent_task_id))
+              << "Task metadata not found in either GCS or lineage cache. It may have been "
+                 "evicted "
+              << "by the redis LRU configuration. Consider increasing the memory "
+                 "allocation via "
+              << "ray.init(redis_max_memory=<max_memory_bytes>).";
+          // Use a copy of the cached task spec to get the parent actor id.
+          Task parent_task = lineage_cache_.GetTaskOrDie(parent_task_id);
+          new_actor_data.set_parent_actor_id(parent_task.GetTaskSpecification().ActorId().Binary());
+          FinishAssignedActorTaskHelper(actor_id, new_actor_data, resumed_from_checkpoint);
+          if(!resumed_from_checkpoint){
+            // The actor was not resumed from a checkpoint. We extend the actor's
+            // frontier as usual since there is no frontier to restore.
+            ExtendActorFrontier(task, actor_id, actor_handle_id);
+          }
+        }));
   }
-
-  if (!resumed_from_checkpoint) {
+  else if (!resumed_from_checkpoint) {
```

Comment:
This looks like it could just be an `else` (you would never have `resumed_from_checkpoint` for regular actor tasks). Also, please move to the previous line.

Reply:
The if condition existed before, hence I stayed with it. Can remove if it's unnecessary.

Reply:
Yeah it used to be necessary because we also wanted to call `ExtendActorFrontier` for actor creation tasks that didn't resume from a checkpoint, but now we're calling that in the callback for the actor table lookup. So we can just make this an `else` now.

Reply:
Done.